go-controller/cmd/ovnkube/ovnkube.go

```diff
@@ -224,15 +279,39 @@ func runOvnKube(ctx *cli.Context, cancel context.CancelFunc) error {
 	}()
 	var masterWatchFactory *factory.WatchFactory
 	var masterEventRecorder record.EventRecorder
-	if master != "" {
+
+	if runMode.networkControllerManager {
 		var err error
 		// create factory and start the controllers asked for
 		masterWatchFactory, err = factory.NewMasterWatchFactory(ovnClientset.GetMasterClientset())
 		if err != nil {
 			return err
 		}
 		defer masterWatchFactory.Shutdown()
+	}
 
+	if runMode.clusterManager {
+		var clusterManagerWatchFactory *factory.WatchFactory
+		if runMode.networkControllerManager {
+			clusterManagerWatchFactory = masterWatchFactory
+		} else {
+			clusterManagerWatchFactory, err = factory.NewClusterManagerWatchFactory(ovnClientset.GetClusterManagerClientset())
+			if err != nil {
+				return err
+			}
+			defer clusterManagerWatchFactory.Shutdown()
+		}
+
+		cm := clustermanager.NewClusterManager(ovnClientset.GetClusterManagerClientset(), clusterManagerWatchFactory,
+			wg, util.EventRecorder(ovnClientset.KubeClient))
+		err = cm.Start(ctx.String("init-network-controller-manager"), ctx.Context)
```

Comment:
can you use the stop channel instead of context? I see in this function that when context is cancelled, stop channel "stop" is closed. This seems to be used to signal stuff to stop in this code. Passing in ctx will work too but do you think we should follow this existing pattern?